osquery/tables/system/darwin/certificates.mm

```diff
@@ -66,17 +51,94 @@ void genCertificate(const SecCertificateRef& SecCert, QueryData& results) {
       (cert->skid) ? genKIDProperty(cert->skid->data, cert->skid->length) : "";
 
   results.push_back(r);
-  X509_free(cert);
+}
+
+void genKeychainCertificate(const SecCertificateRef& SecCert,
+                            QueryData& results) {
+  auto der_encoded_data = SecCertificateCopyData(SecCert);
+  if (der_encoded_data == nullptr) {
+    return;
+  }
+
+  const unsigned char* der_bytes = CFDataGetBytePtr(der_encoded_data);
+  auto length = CFDataGetLength(der_encoded_data);
+  auto cert = d2i_X509(nullptr, &der_bytes, length);
+
+  if (cert != nullptr) {
+    auto path = getKeychainPath((SecKeychainItemRef)SecCert);
+    genCertificate(cert, path, results);
+    X509_free(cert);
+  }
+
   CFRelease(der_encoded_data);
 }
 
+void genFileCertificate(const std::string& path, QueryData& results) {
+  std::string content;
+  auto s = readFile(path, content);
+  if (s.ok()) {
+    const unsigned char* bytes = (const unsigned char*)content.c_str();
+    X509* cert = d2i_X509(nullptr, &bytes, content.size());
+
+    // If cert couldn't be read as DER, attempt
+    // to read it as a PEM.
+    if (cert == nullptr) {
```

Comment:
I'd changed the logic to handle multiple certs in a PEM and accidentally deleted the lines that handle DER certs.